test/compiler/inline.jl

```diff
@@ -1537,6 +1537,34 @@ let
     @test get_finalization_count() == 1000
 end
 
+
+function cfg_finalization7(io)
+    for i = -999:1000
+        o = DoAllocWithField(0)
+        o.x = 0
+        if i == 1000
+            o.x = i # with `setfield!`
+        end
+        o.x = i
+        if i == 999
+            o.x = i
+        end
+        o.x = 0
+        if i == 1000
+            o.x = i
+        end
+    end
+end
```

Comment:
Can't we simplify this case further, or is this code structure necessary? A bit hard to see what's being tested here.